### PR TITLE
Use latest modifiedTime in merger results

### PR DESCRIPTION
**Pull request by jamieparkinson**
Due to the autoformatter going a bit weird (probably from my terrible merge conflict resolutions) this also includes some linting fixes to seldom-touched Python files 🤷‍♂️

### CODE REVIEW

File: common/internal_model/src/test/scala/uk/ac/wellcome/models/work/generators/InstantGenerators.scala [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import uk.ac.wellcome.models.work.internal._
+
+import scala.concurrent.duration._
+import scala.util.Random
+
+trait InstantGenerators {
+
+  val now: Instant = Instant.now
+
+  def createInstantRangeWith(from: String,
```
Comment (kenoir) [inline common/internal_model/src/test/scala/uk/ac/wellcome/models/work/generators/InstantGenerators.scala:14]: Fancy.

File: common/internal_model/src/main/scala/uk/ac/wellcome/models/work/internal/Work.scala [hunk 2]
```diff
@@ -198,8 +201,12 @@ object WorkFsm {
   }
 
   implicit val sourceToMerged = new Transition[Source, Merged] {
-    def state(state: Source, numberOfSources: Int): Merged =
-      Merged(state.sourceIdentifier, numberOfSources)
+    def state(state: Source, args: (Option[Instant], Int)): Merged =
+      Merged(
+        state.sourceIdentifier,
+        args._1.getOrElse(state.modifiedTime),
```
Comment (alicefuzier) [inline common/internal_model/src/main/scala/uk/ac/wellcome/models/work/internal/Work.scala:207]: In what instance is the modifedTime not present?

File: common/internal_model/src/test/scala/uk/ac/wellcome/models/work/generators/ImageGenerators.scala [hunk 3]
```diff
@@ -43,12 +46,14 @@ trait ImageGenerators
       IdState.Identified(createCanonicalId, createSourceIdentifier),
     location: DigitalLocationDeprecated = createDigitalLocation,
     version: Int = 1,
+    modifiedTime: Instant = instantInLast30Days,
```
Comment (alicefuzier) [inline common/internal_model/src/test/scala/uk/ac/wellcome/models/work/generators/ImageGenerators.scala:49]: neat!

File: common/internal_model/src/main/scala/uk/ac/wellcome/models/work/internal/Work.scala (hunk 2 above)
Comment (jamieparkinson) [reply common/internal_model/src/main/scala/uk/ac/wellcome/models/work/internal/Work.scala:207]: This was for convenience if you want to transition the state without changing the `modifiedTime` (eg you use the existing one in the `Source` state) - makes life easier in tests in particular

File: pipeline/id_minter/id_minter_images/src/test/scala/uk/ac/wellcome/platform/id_minter_images/IdMinterFeatureTest.scala [hunk 4]
```diff
@@ -28,7 +28,7 @@ class IdMinterFeatureTest
         withWorkerService(messageSender, queue, identifiersTableConfig) { _ =>
           eventuallyTableExists(identifiersTableConfig)
           val image = createUnmergedImage mergeWith (
-            mergedWork().toSourceWork, None, 1
+            mergedWork().toSourceWork, None, now
```
Comment (alicefuzier) [inline pipeline/id_minter/id_minter_images/src/test/scala/uk/ac/wellcome/platform/id_minter_images/IdMinterFeatureTest.scala:31]: parameter names would be helpful here. I find it a bit hard to follow

File: common/internal_model/src/test/scala/uk/ac/wellcome/models/work/generators/InstantGenerators.scala [hunk 5]
```diff
@@ -0,0 +1,36 @@
+package uk.ac.wellcome.models.work.generators
+
+import java.time.{Instant, LocalDateTime}
+
+import uk.ac.wellcome.models.work.internal._
+
+import scala.concurrent.duration._
+import scala.util.Random
+
+trait InstantGenerators {
```
Comment (alicefuzier) [inline common/internal_model/src/test/scala/uk/ac/wellcome/models/work/generators/InstantGenerators.scala:10]: Really like this trait. I should probably backport it to the transformer tests cause this is much nicer than what I did there

**Review by alicefuzier — APPROVED** [review]
Really nice!

File: common/internal_model/src/main/scala/uk/ac/wellcome/models/work/internal/Work.scala [hunk 6]
```diff
@@ -147,16 +147,18 @@ object WorkState {
...
   case class Merged(
     sourceIdentifier: SourceIdentifier,
-    numberOfSources: Int
+    modifiedTime: Instant,
+    numberOfSources: Int,
   ) extends WorkState {
 
     type WorkDataState = DataState.Unidentified
-    type TransitionArgs = Int
+    type TransitionArgs = (Option[Instant], Int)
```
Comment (warrd) [inline common/internal_model/src/main/scala/uk/ac/wellcome/models/work/internal/Work.scala:155]: Under what circumstances would this be `None`?

Comment (jamieparkinson) [reply common/internal_model/src/main/scala/uk/ac/wellcome/models/work/internal/Work.scala:155]: https://github.com/wellcomecollection/catalogue/pull/962#discussion_r509175537

Comment (jamieparkinson) [reply common/internal_model/src/main/scala/uk/ac/wellcome/models/work/internal/Work.scala:155]: "I want to transition the state, and optionally update the modifiedTime when I do that"

Comment (warrd) [reply common/internal_model/src/main/scala/uk/ac/wellcome/models/work/internal/Work.scala:155]: Ah sorry missed that
